### PR TITLE
{MUSIC Paint} update magic wand with spinbox

### PR DESCRIPTION
**Pull request by mathildemerle**
Update magic wand when we change spin box.

Spinbox are medIntParameter.

Use values from spinbox and sliders.

:m:

### CODE REVIEW

File: src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp [hunk 1]
```diff
@@ -298,7 +294,7 @@ AlgorithmPaintToolBox::AlgorithmPaintToolBox(QWidget *parent ) :
     layout->addLayout( brushSizeLayout );
 
     // Magic wand's widgets
-    m_wandUpperThresholdSlider = new medDoubleParameter("Upper Threshold", this);
+    m_wandUpperThresholdSlider = new medIntParameter("Upper Threshold", this);
```
Comment (LoicCadour) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: Why this change?

**Comment by LoicCadour** [comment]
The pb with the signal *editingFinished* is that it is not emitted when you step up or down the spinBox and when you programatically set the value.
A solution is to use the signal *valueChanged* and disable the keyboard tracking of the spinBox (so that the signal is not emitted three times when typing 123: http://doc.qt.io/qt-4.8/qabstractspinbox.html#keyboardTracking-prop)
I'll push a commit on it.

**Comment by LoicCadour** [comment]
... if I could push on your repo.
Could you check on your side that I have the right accesses?

**Comment by LoicCadour** [comment]
I can't push on your repo (could you check if I have the right access?), but the commit I am talking about is here: https://github.com/LoicCadour/medInria-public/commit/91f087b288868a7d652e66abc3a6f4ff1df80587
Could you put it in this PR?

File: src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp (hunk 1 above)
Comment (mathildemerle) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: Because we do not want double values displayed, just integer :)

**Comment by mathildemerle** [comment]
I pushed your commit loic :)

File: src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp (hunk 1 above)
Comment (LoicCadour) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: And why don't we want that?

Comment (mathildemerle) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: Hum, you're talking about the case of volumes of double? Do you think the case will happened? Indeed if you have these kind of volumes...

Comment (LoicCadour) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: So, in the end, what should we do?

Comment (mathildemerle) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: (i'm on irc if needed) I'm switching back to double :)

**Comment by mathildemerle** [comment]
Increment of spinbox has been put to 1, and we use medDoubleParameter for magic wand

**Comment by LoicCadour** [comment]
Fine.

File: src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp [hunk 2]
```diff
@@ -542,6 +498,8 @@ void AlgorithmPaintToolBox::updateMagicWandComputation()
 
 void AlgorithmPaintToolBox::activateStroke()
 {
+    m_wandInfo->hide();
```
Comment (mathildemerle) [inline src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp]: I've rebased this PR, and added this line which allow to hide the wand label ("Select a pixel in the image blablabla") when we click on Paint/Erase.

**Comment by fcollot** [comment]
ok
